src/server.jl

```diff
@@ -49,35 +87,53 @@ function get_param(source::Vector{HTTP.Forms.Multipart}, name::String, required:
 end
 
 
-function to_param_type(::Type{T}, strval::String) where {T <: Number}
+function to_param_type(::Type{T}, strval::String; stylectx=nothing) where {T <: Number}
     parse(T, strval)
 end
 
-to_param_type(::Type{T}, val::T) where {T} = val
-to_param_type(::Type{T}, ::Nothing) where {T} = nothing
-to_param_type(::Type{String}, val::Vector{UInt8}) = String(copy(val))
-to_param_type(::Type{Vector{UInt8}}, val::String) = convert(Vector{UInt8}, copy(codeunits(val)))
-to_param_type(::Type{Vector{T}}, val::Vector{T}, _collection_format::Union{String,Nothing}) where {T} = val
+to_param_type(::Type{T}, val::T; stylectx=nothing) where {T} = val
+to_param_type(::Type{T}, ::Nothing; stylectx=nothing) where {T} = nothing
+to_param_type(::Type{String}, val::Vector{UInt8}; stylectx=nothing) = String(copy(val))
+to_param_type(::Type{Vector{UInt8}}, val::String; stylectx=nothing) = convert(Vector{UInt8}, copy(codeunits(val)))
+to_param_type(::Type{Vector{T}}, val::Vector{T}, _collection_format::Union{String,Nothing}; stylectx=nothing) where {T} = val
+to_param_type(::Type{Vector{T}}, json::Vector{Any}; stylectx=nothing) where {T} = [to_param_type(T, x; stylectx) for x in json]
+
+function to_param_type(::Type{Vector{T}}, json::Dict{String, Any}; stylectx=nothing) where {T <: APIModel}
+    if !isnothing(stylectx) && is_deep_explode(stylectx)
+        cvt = convert_dict_to_array(json)
+        if isa(cvt, Vector)
+            to_param_type(Vector{T}, cvt; stylectx)
+        else
+            to_param_type(T, json; stylectx)
+        end
+    else
+        to_param_type(T, json; stylectx)
```

Comment:
`to_param_type(::Type{Vector{T}},` should be returning a `Vector{T}`, no?
Should we be throwing an exception instead?

Reply:
I think exception is better and is updated accordingly